eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -365,18 +370,49 @@ function ProcessReplace
     return $replaceMatrix
 }
 
-function ProcessImport([MatrixParameter[]]$matrix, [String]$selection, [Array]$nonSparseParameters, [Hashtable]$displayNamesLookup)
-{
+function ProcessEnvironmentVariableReferences([array]$matrix, $displayNamesLookup) {
+    $updatedMatrix = @()
+
+    foreach ($element in $matrix) {
+        if (!$element || [bool]$element.PSobject.Properties.name -notmatch "_permutation") {
+            continue
+        }
+        $updated = [MatrixParameter[]]@()
+
+        foreach ($perm in $element._permutation) {
+            # Iterate nested permutations or run once for singular values (int, string, bool)
+            foreach ($flattened in $perm.Flatten()) {
+                if ($flattened.Value?.GetType() -eq "".GetType() -and $flattened.Value.StartsWith("env:")) {
+                    $envKey = $flattened.Value.Replace("env:", "")
```

Comment:
Couldn't you use `Get-Item $flattend.Value`" I think that would get you the name and value.